cocos/network/WebSocket.cpp

```diff
@@ -241,21 +323,42 @@ WebSocket::WebSocket()
 , _SSLConnection(0)
 , _wsProtocols(nullptr)
 {
+    // reserve data buffer to avoid allocate memory frequently
+    _receivedData.reserve(WS_RESERVE_RECEIVE_BUFFER_SIZE);
+    if (__websocketInstances == nullptr)
+    {
+        __websocketInstances = new std::vector<WebSocket*>();
+    }
+    
+    __websocketInstances->push_back(this);
 }
 
 WebSocket::~WebSocket()
 {
-    close();
-    CC_SAFE_RELEASE_NULL(_wsHelper);
-    
-    if(_wsProtocols)
+    LOGD("In the destructor of WebSocket (%p)\n", this);
+    CC_SAFE_DELETE(_wsHelper);
+
+    if (_wsProtocols != nullptr)
     {
         for (int i = 0; _wsProtocols[i].callback != nullptr; ++i)
         {
             CC_SAFE_DELETE_ARRAY(_wsProtocols[i].name);
         }
     }
-	CC_SAFE_DELETE_ARRAY(_wsProtocols);
+    CC_SAFE_DELETE_ARRAY(_wsProtocols);
+    
+    if (__websocketInstances != nullptr)
+    {
+        auto iter = std::find(__websocketInstances->begin(), __websocketInstances->end(), this);
+        if (iter != __websocketInstances->end())
+        {
+            __websocketInstances->erase(iter);
+        }
+        else
+        {
+            LOGD("ERROR: WebSocket instance (%p) added to container!\n", this);
```

Comment:
To be more clear
ERROR: WebSocket instance (%p) haven't been added to container!